crates/core/src/client/config.rs

```diff
@@ -4,31 +4,56 @@ use starknet::providers::jsonrpc::{HttpTransport, JsonRpcTransport};
 use starknet::providers::JsonRpcClient;
 use url::Url;
 
+use super::constants::{KATANA_RPC_URL, MADARA_RPC_URL};
 use super::errors::ConfigError;
 
 fn get_env_var(name: &str) -> Result<String, ConfigError> {
     std::env::var(name).map_err(|_| ConfigError::EnvironmentVariableMissing(name.into()))
 }
 
-#[derive(Default)]
+#[derive(Default, Clone)]
+pub enum Network {
+    #[default]
+    Katana,
+    Madara,
+    Mainnet,
+    Goerli1,
+    Goerli2,
+    ProviderUrl(Url),
+}
+
+#[derive(Default, Clone)]
 /// Configuration for the Starknet RPC client.
 pub struct StarknetConfig {
-    /// Rpc url.
-    pub url: String,
+    /// Additional configuration if the underlying provider is a Sequencer provider.
+    pub network: Network,
     /// Kakarot contract address.
     pub kakarot_address: FieldElement,
     /// Proxy account class hash.
     pub proxy_account_class_hash: FieldElement,
 }
 
 impl StarknetConfig {
-    pub fn new(url: String, kakarot_address: FieldElement, proxy_account_class_hash: FieldElement) -> Self {
-        StarknetConfig { url, kakarot_address, proxy_account_class_hash }
+    pub fn new(network: Network, kakarot_address: FieldElement, proxy_account_class_hash: FieldElement) -> Self {
+        StarknetConfig { network, kakarot_address, proxy_account_class_hash }
     }
 
     /// Create a new `StarknetConfig` from environment variables.
     pub fn from_env() -> Result<Self, ConfigError> {
-        let starknet_rpc = get_env_var("STARKNET_RPC_URL")?;
+        let network = get_env_var("STARKNET_NETWORK")?;
+        let network = match network.to_lowercase().as_str() {
+            // TODO: Add possibility to set url for katana and madara in env rather than constants.
+            "katana" => Network::Katana,
+            "madara" => Network::Madara,
+            // TODO: Add possibility to override gateway url for mainnet and testnet.
```

Comment:
I think that it's not required, and I don't see any situation where it could be used. There is only one gateway per network

Reply:
but it shouldn't be here then, mb!

Reply:
fixed